src/index/mod.rs

```diff
@@ -1,27 +1,40 @@
-use std::collections::{HashMap, HashSet};
+pub mod cache;
 use std::path::{Path, PathBuf};
+
 use std::time::SystemTime;
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
 
 use canonical_path::CanonicalPathBuf;
 use walkdir::{DirEntry, WalkDir};
 
 use anyhow::Error;
 use log;
+use nom::{sequence::preceded, IResult};
 
 use crate::id::ResourceId;
 use crate::meta::ResourceMeta;
 
+pub const INDEX_CACHE_DIR: &'static str = ".ark/index/cache";
```

Comment:
It's not cache, just persistence. Index and cache have many similarities, but cache of index doesn't make sense. Only `.ark/index` is needed.